packages/edit-site/src/hooks/commands/use-common-commands.js

```diff
@@ -244,11 +244,17 @@ function useGlobalStylesOpenRevisionsCommands() {
 	const isMobileViewport = useViewportMatch( 'medium', '<' );
 	const isEditorPage = ! getIsListPage( params, isMobileViewport );
 	const history = useHistory();
-	const hasRevisions = useSelect(
-		( select ) =>
-			select( coreStore ).getCurrentThemeGlobalStylesRevisions()?.length,
-		[]
-	);
+	const hasRevisions = useSelect( ( select ) => {
+		const { getRevisions, __experimentalGetCurrentGlobalStylesId } =
+			select( coreStore );
+		const globalStylesRevisions = getRevisions(
+			'root',
+			'globalStyles',
+			__experimentalGetCurrentGlobalStylesId()
+		);
+		return !! globalStylesRevisions?.length;
```

Comment:
Thanks for picking this up @t-hamano 🙇🏻 

Because the revisions response could be larger (at the moment up to 100 items in the collection), maybe it would be faster to grab the revision count from the `version-history` meta?

E.g., 

```js
			const globalStylesId = __experimentalGetCurrentGlobalStylesId();
			const globalStyles = globalStylesId
				? getEntityRecord( 'root', 'globalStyles', globalStylesId )
				: undefined;
			return globalStyles?._links?.[ 'version-history' ]?.[ 0 ]?.count || 0;
```

Reply:
+1, thanks for working on this!

That would be good if it's possible to use `version-history` and avoid the extra API call 👍

Reply:
Indeed. I updated it with [2257407](https://github.com/WordPress/gutenberg/pull/56738/commits/2257407da0407bf3a94214df177a4baf61626453) and confirmed that it continues to work as expected 👍